packages/components/src/utils/getParsedDate.ts

```diff
@@ -0,0 +1,39 @@
+import { isMatch, parse } from "date-fns"
+
+const SUPPORTED_DATE_FORMATS = [
+  "dd/MM/yyyy",
+  "d MMM yyyy",
+  "d MMMM yyyy",
+  "dd MMM yyyy",
+  "dd MMMM yyyy",
+  "yyyy-MM-dd",
+  "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+]
+
+export const getParsedDate = (dateString: string) => {
```

Comment:
actually the flow i had in mind was `reduce(function, new Date()`

then in our reducer function:
```ts
if (isMatch(date)) { // parse } 
```

and we can put supported date formats in increasing order of priority, so that the code looks cleaner overall. not a big deal though, just that this was not quite what i had in mind

Reply:
Hmm I a bit confused, I think that's what I'm doing right now? The reducer function is indeed going by priority (from first to last element) and it starts with `undefined` to differentiate between whether the date string has been parsed or not. If parsed then the type is of Date, otherwise it will remain as undefined.